rocAL/rocAL_pybind/amd/rocal/plugin/tf.py

```diff
@@ -22,40 +22,46 @@
 import cupy as cp
 import rocal_pybind as b
 import amd.rocal.types as types
+
+
 class ROCALGenericImageIterator(object):
     def __init__(self, pipeline):
         self.loader = pipeline
-        self.w = b.getOutputWidth(self.loader._handle)
-        self.h = b.getOutputHeight(self.loader._handle)
-        self.n = b.getOutputImageCount(self.loader._handle)
-        color_format = b.getOutputColorFormat(self.loader._handle)
-        self.p = (1 if (color_format == int(types.GRAY)) else 3)
-        height = self.h*self.n
-        self.out_tensor = None
-        self.out_image = np.zeros((height, self.w, self.p), dtype = "uint8")
+        self.output_list = None
         self.bs = pipeline._batch_size
 
     def next(self):
         return self.__next__()
 
     def __next__(self):
-        if b.getRemainingImages(self.loader._handle) < self.bs:
```

Comment:
Is it not necessary to check remaining images

Reply:
Its been checked as part of the rocalRun API